### PR TITLE
Report meaningful line numbers for inline script errors

### LINKED ISSUE #12744: Correctly set the line offset of inline scripts
We currently [set the offset of a script as 0 when its loaded inline](https://github.com/servo/servo/blob/0fc0db67c64bf8ab4593d1f84b8d1be517fa1b74/components/script/dom/window.rs#L949).

This means that if an inline script errors, the line number is given relative to the start of the script, not the start of the html file it's in. This is less useful 😄 

To fix this, we'll probably need to hook into the parser somehow, not sure. There doesn't seem to be any line number data stored in Servo directly.

(Perhaps have create_element take the line number?)

### PR DESCRIPTION
**Pull request by jdm**
Rebased from #14661.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #12744 and partially #9604
- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14963)

### CODE REVIEW

File: components/script/dom/element.rs [hunk 1]
```diff
@@ -145,10 +145,25 @@ impl fmt::Debug for Element {
...
 pub enum ElementCreator {
-    ParserCreated,
+    ParserCreated(u64),
     ScriptCreated,
 }
 
+impl ElementCreator {
+    pub fn is_parser_created(&self) -> bool {
+        match *self {
+            ElementCreator::ParserCreated(l) => true,
```
Comment (asajeffrey) [inline components/script/dom/element.rs]: Nit: shouldn't this be a `_` pattern?

File: components/script/dom/htmlscriptelement.rs [hunk 2]
```diff
@@ -508,7 +512,7 @@ impl HTMLScriptElement {
         let window = window_from_node(self);
         rooted!(in(window.get_cx()) let mut rval = UndefinedValue());
         window.upcast::<GlobalScope>().evaluate_script_on_global_with_result(
-            &script.text, script.url.as_str(), rval.handle_mut());
+            &script.text, script.url.as_str(), rval.handle_mut(), self.line_number as u32);
```
Comment (asajeffrey) [inline components/script/dom/htmlscriptelement.rs]: Can we avoid this cast?

**Review by asajeffrey — APPROVED** [review]
Looks good, just some nits.

File: components/script/dom/htmlscriptelement.rs (hunk 2 above)
Comment (jdm) [inline components/script/dom/htmlscriptelement.rs]: We made the line numbers u64 in html5ever, so there's going to be a cast _somewhere_ until we change that.

Comment (asajeffrey) [inline components/script/dom/htmlscriptelement.rs]: Oh well.

**Comment by asajeffrey** [comment]
```
error[E0308]: mismatched types
  --> /Users/servo/buildbot/slave/mac-rel-wpt2/build/components/script/dom/htmlstyleelement.rs:53:51
   |
53 |             parser_inserted: Cell::new(creator == ElementCreator::ParserCreated),
   |                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `dom::element::ElementCreator`, found fn item
   |
   = note: expected type `dom::element::ElementCreator`
   = note:    found type `fn(u64) -> dom::element::ElementCreator {dom::element::ElementCreator::ParserCreated}`

error[E0308]: mismatched types
  --> /Users/servo/buildbot/slave/mac-rel-wpt2/build/components/script/dom/htmlstyleelement.rs:54:61
   |
54 |             in_stack_of_open_elements: Cell::new(creator == ElementCreator::ParserCreated),
   |                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `dom::element::ElementCreator`, found fn item
   |
   = note: expected type `dom::element::ElementCreator`
   = note:    found type `fn(u64) -> dom::element::ElementCreator {dom::element::ElementCreator::ParserCreated}`

```

**Comment by KiChjang** [comment]
Looking great! You'll need to update the test expectations for the following two passing tests:
```
  ▶ Unexpected subtest result in /html/webappapis/scripting/processing-model-2/window-onerror-parse-error.html:
  └ PASS [expected FAIL] correct line number passed to window.onerror

  ▶ Unexpected subtest result in /html/webappapis/scripting/processing-model-2/window-onerror-runtime-error.html:
  └ PASS [expected FAIL] correct line number passed to window.onerror
```

**Pull request merged**